code/world/initialization/0_preMapLoad.dm

```diff
@@ -217,12 +217,14 @@
 	return
 
 /proc/buildMaterialCache()
-	material_cache = list()
+	material_cache_matid = list()
+	material_cache_type = list()
 	var/materialList = concrete_typesof(/datum/material)
 	for(var/datum/material/mat as anything in materialList)
 		if(initial(mat.cached))
 			var/datum/material/M = new mat()
-			material_cache[M.getID()] = M.getImmutable()
+			material_cache_matid[M.getID()] = M.getImmutable()
+			material_cache_type[M.type] = M.getImmutable()
```

Comment:
If you are doing it by both, the second `getImmutable()` will create a second copy. Better to do this instead.
```suggestion
			material_cache_type[M.type] = material_cache_matid[M.getID()]
```

Reply:
I did this and then it turned out to no longer be needed since i ripped out the string matids.